File: backend/src/main/java/com/herd/squire/models/attributes/Misc.java
package com.herd.squire.models.attributes;

import com.herd.squire.models.powers.ModifierCategory;

public class Misc extends Attribute {
    private ModifierCategory modifierCategory;
    private boolean characteristicDependant;

    public Misc() {
        super();
        this.modifierCategory = ModifierCategory.MISC;
    }

    public Misc(String id, String name, String description, int sid, boolean author, int version, ModifierCategory modifierCategory, boolean characteristicDependant) {
        super(id, name, description, AttributeType.MISC, sid, author, version);
        this.modifierCategory = modifierCategory;
        this.characteristicDependant = characteristicDependant;
    }

    public ModifierCategory getModifierCategory() {
        return modifierCategory;
    }

    public void setModifierCategory(ModifierCategory modifierCategory) {
        this.modifierCategory = modifierCategory;
    }

    public boolean isCharacteristicDependant() {
        return characteristicDependant;
    }

    public void setCharacteristicDependant(boolean characteristicDependant) {
        this.characteristicDependant = characteristicDependant;
    }
}

File: frontend/src/app/shared/models/attributes/misc.ts
import {Attribute} from './attribute';
import {ModifierCategory} from '../modifier-category.enum';
import {AttributeType} from './attribute-type.enum';

export class Misc extends Attribute {
  type = 'Misc';
  modifierCategory: ModifierCategory = ModifierCategory.MISC;
  characteristicDependant = false;

  constructor() {
    super();
    this.attributeType = AttributeType.MISC;
  }
}
